From benh@kernel.crashing.org Wed Dec 27 20:27:08 2006
From: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Date: Thu, 28 Dec 2006 15:26:59 +1100
Subject: USB: Fix OHCI warning
To: akpm@osdl.org
Cc: Greg KH <greg@kroah.com>
Message-ID: <1167280019.23340.55.camel@localhost.localdomain>
Content-Type: text/plain


This patch fixes a warning introduces by the split endian OHCI support
patch on platforms that don't have readl_be/writel_be variants (though
mostly harmless as those are called in an if (0) statement, but gcc
still warns).

Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/ohci.h |    8 ++++++++
 1 file changed, 8 insertions(+)

--- gregkh-2.6.orig/drivers/usb/host/ohci.h
+++ gregkh-2.6/drivers/usb/host/ohci.h
@@ -505,17 +505,25 @@ static inline struct usb_hcd *ohci_to_hc
 static inline unsigned int _ohci_readl (const struct ohci_hcd *ohci,
 					__hc32 __iomem * regs)
 {
+#ifdef CONFIG_USB_OHCI_BIG_ENDIAN_MMIO
 	return big_endian_mmio(ohci) ?
 		readl_be ((__force u32 *)regs) :
 		readl ((__force u32 *)regs);
+#else
+	return readl ((__force u32 *)regs);
+#endif
 }
 
 static inline void _ohci_writel (const struct ohci_hcd *ohci,
 				 const unsigned int val, __hc32 __iomem *regs)
 {
+#ifdef CONFIG_USB_OHCI_BIG_ENDIAN_MMIO
 	big_endian_mmio(ohci) ?
 		writel_be (val, (__force u32 *)regs) :
 		writel (val, (__force u32 *)regs);
+#else
+		writel (val, (__force u32 *)regs);
+#endif
 }
 
 #ifdef CONFIG_ARCH_LH7A404
